### PR TITLE
feat: add `iter/cuevery`

### LINKED ISSUE #2332: [RFC]: add `@stdlib/iter/cuevery`
### Description

This RFC proposes adding the package `@stdlib/iter/cuevery`, which cumulatively tests whether every iterated value is truthy. The function should be a transform iterator, continuing to iterate while source iterator values are available.

```javascript
var array2iterator = require( '@stdlib/array/to-iterator' );

var arr = array2iterator( [ true, true, true, false, true ] );

var it = iterCuEvery( arr );

var v = it.next().value;
// returns true

v = it.next().value;
// returns true

v = it.next().value;
// returns true

v = it.next().value;
// returns false

v = it.next().value;
// returns false

var bool = it.next().done;
// returns true
```

### Related Issues

No.

### Questions

No.

### Other

- See also `@stdlib/iter/every`

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Dipti0704**
Resolves #2332.

## Description

> What is the purpose of this pull request?

This pull request:

-  Implements a new iterator function `iterCuevery`
- Adds functionality to cumulatively test whether every iterated value is truthy
- Enhances the @stdlib/iter module ecosystem

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #2332 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

- Input Iterator: The function takes an input iterator.
- Output Iterator: Returns a new iterator that yields boolean values.
- Yielding True: Continues yielding true while all values are falsy.
- Yielding False: Yields false once a truthy value is encountered and for all subsequent iterations.
- Handling Input: Handles both iterable and non-iterable input iterators.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Dipti0704** [comment]
@kgryte can you review this?

File: lib/node_modules/@stdlib/_tools/repl-txt/rules/interface-spacing/lib/main.js
Comment (kgryte) [inline lib/node_modules/@stdlib/_tools/repl-txt/rules/interface-spacing/lib/main.js:1]: This file should not be modified.

File: lib/node_modules/@stdlib/iter/cuevery/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/README.md:1]: This file is empty.

File: lib/node_modules/@stdlib/iter/cuevery/docs/repl.txt
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/docs/repl.txt:1]: This file does not conform to our REPL text conventions. Please read the REPL text guide.

File: lib/node_modules/@stdlib/iter/cuevery/docs/types/index.d.ts
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/docs/types/index.d.ts:1]: Missing license header.

File: lib/node_modules/@stdlib/iter/cuevery/docs/types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+* v = it.next().value;
+* // returns false
+*
+* v = it.next().value;
+* // returns false
+*
+* v = it.next().done;
+* // returns true
+*/
+declare function iterCuEvery( iterator: Iterator ): Iterator;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/docs/types/index.d.ts:40]: You can improve return value specificity by using the `TypedIterator` interface.

File: lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import iterCuEvery from './index';
+//import isIteratorLike from '@stdlib/assert/is-iterator-like';
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts:20]: Why is this here?

File: lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            return { value: value ?? true, done: true };
+        }
+    };
+}
+
+// TESTS //
+
+// The function returns an iterator...
+{
+    iterCuEvery(iterator()); // $ExpectType Iterator
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts:43]: Incorrect spacing and indentation throughout this file.

File: lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+// The compiler throws an error if the function is provided a value other than an iterator protocol-compliant object...
+{
+    iterCuEvery(5 as any); // $ExpectError
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts:48]: Why are you casting?

File: lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    iterCuEvery(true as any); // $ExpectError
+    iterCuEvery(null as any); // $ExpectError
+    iterCuEvery(undefined as any); // $ExpectError
+    iterCuEvery([] as any); // $ExpectError
+    iterCuEvery({} as any); // $ExpectError
+    //iterCuEvery((x: number): number => x as any as Iterator); // $ExpectError
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts:54]: Why is this commented out?

File: lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+// Test iterator with Symbol.iterator if supported
+{
+    const it = iterator();
+    if (Symbol.iterator in it) {
+        iterCuEvery(it); // $ExpectType Iterator
+    }
+}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/docs/types/test.ts:63]: This should be removed.

File: lib/node_modules/@stdlib/iter/cuevery/examples/index.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+var iterMap = require( '@stdlib/iter-map' );
+var iterCuEvery = require( './../lib' );
+
+// Create an iterator which generates uniformly distributed pseudorandom numbers:
+var rand = randu({
+    'iter': 100
+});
+
+// Create an iterator which applies a threshold to generated numbers:
+var it = iterMap( rand, function threshold( r ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/examples/index.js:31]: Do not use function expressions.

File: lib/node_modules/@stdlib/iter/cuevery/examples/index.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+'use strict';
+
+var randu = require( '@stdlib/random-iter-randu' );
+var iterMap = require( '@stdlib/iter-map' );
+var iterCuEvery = require( './../lib' );
+
+// Create an iterator which generates uniformly distributed pseudorandom numbers:
+var rand = randu({
+    'iter': 100
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/examples/index.js:27]: This file has incorrect indentation.

File: lib/node_modules/@stdlib/iter/cuevery/index.js
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/index.js:1]: This file is empty.

Comment (kgryte) [reply lib/node_modules/@stdlib/iter/cuevery/index.js:1]: Actually, this file should be removed.

File: lib/node_modules/@stdlib/iter/cuevery/lib/index.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+'use strict';
+
+/**
+* Create an iterator which cumulatively tests whether every iterated value is truthy.
+*
+* @module @stdlib/iter-cuevery
+*
+* @example
+* var array2iterator = require( '@stdlib/array-to-iterator' );
+* var iterCuEvery = require( '@stdlib/iter-cuevery' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/lib/index.js:28]: Incorrect module names. See other packages within the monorepo.

File: lib/node_modules/@stdlib/iter/cuevery/lib/main.js [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+* // returns false
+*
+* v = it.next().value;
+* // returns false
+*
+* var bool = it.next().done;
+* // returns true
+*/
+
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/lib/main.js:64]: Why are these spaces here?

File: lib/node_modules/@stdlib/iter/cuevery/lib/main.js [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+function iterCuEvery(iterator) {
+    if (!isIteratorLike(iterator)) {
+        throw new TypeError(format('invalid argument. Must provide an iterator. Value: `%s`.', iterator));
+    }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/lib/main.js:68]: Incorrect spacing and indentation. It is clear that you failed to setup your local development environment, as our linting would have caught these errors. Please read our contributing guidelines.

File: lib/node_modules/@stdlib/iter/cuevery/lib/main.js [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+        if (v.done) {
+            done = true;
+            return {
+                value: FLG,
+                done: true
+            };
+        }
+        if (!v.value) {
+            FLG = false;
+            done = true; // Short-circuit once a falsy value is found
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/lib/main.js:101]: This is not correct behavior. This is a cumulative iterator and should have the same length as the source iterator.

File: lib/node_modules/@stdlib/iter/cuevery/lib/main.js [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+        return: finish
+    };
+
+    if (iteratorSymbol && isFunction(iterator[iteratorSymbol])) {
+        setReadOnly(iter, iteratorSymbol, factory);
+    }
+
+    return iter;
+
+    function next() {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/lib/main.js:84]: You're missing JSDoc comments.

File: lib/node_modules/@stdlib/iter/cuevery/lib/main.js [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+
+function iterCuEvery(iterator) {
+    if (!isIteratorLike(iterator)) {
+        throw new TypeError(format('invalid argument. Must provide an iterator. Value: `%s`.', iterator));
+    }
+
+    var FLG = true; // Assume all values are truthy initially
+    var done = false;
+
+    var iter = {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/lib/main.js:73]: This isn't how we do variable declarations in source files.

File: lib/node_modules/@stdlib/iter/cuevery/lib/main.js [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+                done: true
+            };
+        }
+        return {
+            value: FLG,
+            done: false
+        };
+    }
+
+    function finish(value) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/lib/main.js:113]: JSDoc comments.

File: lib/node_modules/@stdlib/iter/cuevery/lib/main.js [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+                done: true
+            };
+        }
+        return {
+            value: FLG,
+            done: true
+        };
+    }
+
+    function factory() {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/lib/main.js:127]: JSDoc comments.

File: lib/node_modules/@stdlib/iter/cuevery/lib/main.js [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+        return iterCuEvery(iterator[iteratorSymbol]());
+    }
+}
+
+// EXPORTS //
+
+module.exports = iterCuEvery;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/lib/main.js:134]: You did not setup EditorConfig per the contributing guidelines.

File: lib/node_modules/@stdlib/iter/cuevery/package-lock.json
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/package-lock.json:1]: Why is this file here? Remove.

File: lib/node_modules/@stdlib/iter/cuevery/package.json [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+    "name": "The Stdlib Authors",
+    "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
+  },
+  "contributors": [
+    {
+      "name": "The Stdlib Authors",
+      "url": "https://github.com/stdlib-js/stdlib/graphs/contributors"
+    }
+  ],
+  "license": "ISC",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/package.json:40]: What is this? This `package.json` is not how we do things. See other packages.

File: lib/node_modules/@stdlib/iter/cuevery/test/test.js [hunk 19]
```diff
@@ -0,0 +1,129 @@
...
+// * Unless required by applicable law or agreed to in writing, software
+// * distributed under the License is distributed on an "AS IS" BASIS,
+// * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// * See the License for the specific language governing permissions and
+// * limitations under the License.
+// */
+
+
+
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/cuevery/test/test.js:21]: Why are you commenting out our license header?

Comment (kgryte) [reply lib/node_modules/@stdlib/iter/cuevery/test/test.js:21]: This file does not follow project conventions.

File: lib/node_modules/@stdlib/iter/package.json
Comment (kgryte) [inline lib/node_modules/@stdlib/iter/package.json:1]: Why did you delete this file?

**Review by kgryte — CHANGES_REQUESTED** [review]
This PR needs significant modifications before it can be reviewed again and considered for inclusion in the project.

**Pull request closed without merging**